### ISSUE TITLE
Investigate systemd session management

### ISSUE THREAD

**Issue by davidmhewitt** [elementary/session-settings#17]
Just writing down some considerations based on the fact we're retiring cerbere and switching to `gnome-session` to manage session components (like plank and wingpanel).

`gnome-session` >= 3.34 is using systemd units for session components by default and it looks like we're getting that version or newer in 20.04, so we should probably look into this.

Thanks to @quequotion [we know](https://github.com/elementary/cerbere/issues/36#issuecomment-573627648) that we don't **need** to use this new systemd method and we can use the `--builtin` flag on `gnome-session` to carry on supporting the `X-GNOME-AutoRestart` key that we're planning to use for 3.28, but maybe systemd will bring us some advantages like more configurable aggressiveness of restarting components etc...

I imagine that the `--builtin` option will become less and less supported as time goes on, but at least we know we have two options outside of Cerbere for 6.0.

Some info here:
 * https://blogs.gnome.org/benzea/2019/10/01/gnome-3-34-is-now-managed-using-systemd/
 * https://www.youtube.com/watch?v=pdwi3NWAW7I

**Comment by davidmhewitt** [comment]
For the time being to get the daily focal builds working, we've opted to add the `--builtin` option. However, I think there's a desire to switch to systemd session management, will track in a separate issue here:
https://github.com/elementary/os/issues/293

**Comment by quequotion** [comment]
While we're at it could we revisit #4? I never got an answer as to why the changes were reverted.

I note that the startups already have the necessary fields to be started and managed by gnome-session without being listed as RequiredComponents:

```
OnlyShowIn=Pantheon;
NoDisplay=true
X-GNOME-Autostart-Phase=Initialization
X-GNOME-Autostart-Notify=true
X-GNOME-AutoRestart=true
```

Eventually I suspect GNOME will transition their own xdg autostarts to systemd, but I do not as yet know if they are planning a complete conversion of them to systemd services or to use gnome-session as an interface between systemd and xdg startups for backward compatibility.

Listing them as RequiredComponents only ensures that some users get dropped to that confounding "Oops..." screen which they are not going to know what to do with; removing them will do no harm at the moment and may even be necessary anyway once the transition to systemd session management is complete for g-s-d.

**Comment by Anachron** [comment]
Please keep in mind that some distros with non-systemd suppoprt still need some solution outside the systemd world!

**Comment by worldofpeace** [comment]
> Please keep in mind that some distros with non-systemd suppoprt still need some solution outside the systemd world!

Yep, that is why GNOME hasn't dropped any of their xdg autostarts, and claimed they should continue to work. I agree that these should not be removed from session-settings.

**Comment by danirabbit** [comment]
I guess this can be closed since we're now handling both the X and Wayland sessions with systemd session management
